profiling/src/profiling/mod.rs

```diff
@@ -346,6 +346,12 @@ impl TimeCollector {
         profiles: &mut HashMap<ProfileIndex, InternalProfile>,
         started_at: &WallTime,
     ) {
+        if message.key.sample_types.len() == 0 {
+            // profiling disabled, this should not happen!
+            warn!("You spot a bug in the profiler, please be so kind and report this do Datadog.");
```

Comment:
We should improve this message if we are going to emit it. Maybe...

```suggestion
            warn!("A sample with no sample types was recorded in the profiler. Please report this to Datadog.");
```